Packages/com.unity.inputsystem/InputSystem/Plugins/EnhancedTouch/EnhancedTouchSupport.cs

```diff
@@ -0,0 +1,120 @@
+using UnityEngine.InputSystem.Utilities;
+
+////TODO: gesture support
+////TODO: mouse/touch simulation support
+////TODO: high-frequency touch support
+
+////REVIEW: have TouchTap, TouchSwipe, etc. wrapper MonoBehaviours like LeanTouch?
+
+namespace UnityEngine.InputSystem.Touch
+{
+    [AddComponentMenu("Input/Enhanced Touch Support")]
+    public class EnhancedTouchSupport : MonoBehaviour
```

Comment:
So the only reason to have this as a MonoBehaviour is that you can stuff it somewhere in your scene to enable it. Is that a good use of a component? Should we only have one way to enable it (by calling the static `Enable` function)? Or should we maybe add a setting to enable it?

Reply:
Also, should it be `Touch.Enable()` instead? Since all the interaction with the system happens through the `Touch` class? Feels odd to have an `EnhancedTouchSupport` class which has the only purpose of enabling the APIs in the `Touch` class.

Reply:
>Is that a good use of a component?
IMO for prototyping this is the kind of stuff that is super convenient. LeanTouch has a number of MonoBehaviours that are similar in nature and probably wouldn't be deemed "good use of a component". But fact is, they are immensely convenient to use.

Reply:
I guess we could come up with a ton of "super convenient" MonoBehaviours for all our public APIs, but we don't usually ship any of those, but instead let users make them for themselves (in this case it is literally calling a link of code in your enable function). So I'd prefer not to have this (but will let you ultimately decide).

Reply:
Any thoughts on my second comment?

Reply:
Ok, looking at this again, I'm fine not making `EnhancedTouchSupport` a `MonoBehaviour`.

Can see the point RE `Touch` vs `EnhancedTouchSupport` but somewhat on the fence. Just that

```
EnhancedTouchSupport.Enable();
```

so nicely communicates what it does vs

```
Touch.Enable();
```

It's not that touch is disabled, it's that the "EnhancedTouch" subsystem is not enabled by default. But maybe I'm splitting hairs here. Can go and roll it into one.

Should the namespace be `UnityEngine.InputSystem.Touch` or `UnityEngine.InputSystem.EnhancedTouch` BTW? The latter would also get rid of the `Touch.Touch` namespace vs class conflict.

Reply:
I'd lean towards `UnityEngine.InputSystem.EnhancedTouch`.